src/Data/Constraint/Extras/TH.hs

```diff
@@ -19,13 +20,20 @@ deriveArgDict n = do
   let xs = flip map ts $ \case
         Left t -> AppT (AppT (ConT ''ConstraintsFor) t) (VarT c)
         Right t -> (AppT (VarT c) t)
-      l = length xs
-      constraints = foldl AppT (TupleT l) xs
+      constraints = batchConstraints xs
   [d| instance ArgDict $(varT c) $(pure typeHead) where
         type ConstraintsFor  $(pure typeHead) $(varT c) = $(pure constraints)
         argDict = $(LamCaseE <$> matches c constrs 'argDict)
     |]
 
+batchConstraints :: [Type] -> Type
+batchConstraints cs
+  | length cs == 0 = TupleT 0
+  | otherwise = AppT (AppT (TupleT 2) initialBatch) (batchConstraints rest)
+  where
+    (initial, rest) = splitAt (maxTupleSize - 1) cs
+    initialBatch = foldl AppT (TupleT (length initial)) initial
+
```

Comment:
I think it might be better to do something like
```suggestion
batchConstraints :: [Type] -> Type
batchConstraints cs
  | length cs <= maxTupleSize = foldl AppT (TupleT (length cs)) cs
  | otherwise = batchConstraints $ batchConstraints <$> chunkBy maxTupleSize cs
```
maybe there is an issue, but that seems about right. I am out of time to think about this ATM.

Reply:
That seemed to work and I'm very happy to make the change. I couldn't find a `chunkBy`, or any function `:: Int -> [a] -> [[a]]`, in `base` or an existing dependency, so I just wrote an implementation in this library rather than add a new library dependency.